include/swift/Frontend/Frontend.h

```diff
@@ -433,6 +444,65 @@ class CompilerInstance {
   /// Frees up the ASTContext and SILModule objects that this instance is
   /// holding on.
   void freeContextAndSIL();
+
+private:
+  bool loadStdlibAndMaybeSwiftOnoneSupport();
+  ModuleDecl *importUnderlyingModule(ClangImporter *clangImporter);
+  ModuleDecl *importBridgingHeader(ClangImporter *clangImporter);
+
+  void fillInModulesToImportFromImplicitImportModuleNames(
+      SmallVectorImpl<ModuleDecl *> &importModules);
+  
+public: // for static functions in Frontend.cpp
+  
+  struct ImplicitImports {
+    SourceFile::ImplicitModuleImportKind kind;
+    ModuleDecl *objCModuleUnderlyingMixedFramework;
+    ModuleDecl *headerModule;
+    SmallVector<ModuleDecl *, 4> modules;
+    
+    ImplicitImports(CompilerInstance &compiler);
+  };
+  
+private:
+  void createREPLFileWithImports(ImplicitImports &implicitImports);
+  DelayedParsingCallbacks *computeDelayedParsingCallback();
+
+  void createMainFileWithImportsAndAddToModule(ImplicitImports &implicitImports);
+ 
+  void parseALibraryFile(
+      unsigned BufferID,
+      ImplicitImports &implicitImports,
+      PersistentParserState &PersistentState,
+      DelayedParsingCallbacks *DelayedParseCB);
+
+  /// Return true if had load error
+  bool parsePartialModulesAndLibraryFiles(
+      ImplicitImports &implicitImports,
+      PersistentParserState &PersistentState,
+      DelayedParsingCallbacks *DelayedParseCB);
+
+  void
+  checkTypesWhileParsingMain(PersistentParserState &PersistentState,
+                                     DelayedParsingCallbacks *DelayedParseCB);
```

Comment:
Better, but still unhappy, as discussed in the reply to the original comment. (Also, funky indentation here.)

Reply:
Not sure what to do here, since my goal is to group together all the type-checking operations, but apparently parsing main is intimately intertwined with type-checking. I think I need more guidance. Regarding the indentation, I will try to run git-clang-format more often.